generator/internal/descriptor_utils.cc

```diff
@@ -244,27 +244,54 @@ void SetMethodSignatureMethodVars(
   }
 }
 
-void SetResourceRoutingMethodVars(
+void SetHttpResourceRoutingMethodVars(
     google::protobuf::MethodDescriptor const& method,
     VarsDictionary& method_vars) {
-  auto result = ParseResourceRoutingHeader(method);
-  if (result.has_value()) {
-    method_vars["method_request_url_path"] = result->url_path;
-    method_vars["method_request_url_substitution"] = result->url_substitution;
-    std::string param = result->param_key;
-    method_vars["method_request_param_key"] = param;
-    std::vector<std::string> chunks;
-    auto const* input_type = method.input_type();
-    for (auto const& sv : absl::StrSplit(param, '.')) {
-      auto const chunk = std::string(sv);
-      auto const* chunk_descriptor = input_type->FindFieldByName(chunk);
-      chunks.push_back(FieldName(chunk_descriptor));
-      input_type = chunk_descriptor->message_type();
+  struct HttpInfoVisitor {
+    HttpInfoVisitor(google::protobuf::MethodDescriptor const& method,
+                    VarsDictionary& method_vars)
+        : method(method), method_vars(method_vars) {}
+
+    void operator()(HttpExtensionInfo const& info) {
+      method_vars["method_request_url_path"] = info.url_path;
+      method_vars["method_request_url_substitution"] = info.url_substitution;
+      std::string param = info.request_field_name;
+      method_vars["method_request_param_key"] = param;
+      std::vector<std::string> chunks;
+      auto const* input_type = method.input_type();
+      for (auto const& sv : absl::StrSplit(param, '.')) {
+        auto const chunk = std::string(sv);
+        auto const* chunk_descriptor = input_type->FindFieldByName(chunk);
+        chunks.push_back(FieldName(chunk_descriptor));
+        input_type = chunk_descriptor->message_type();
+      }
+      method_vars["method_request_param_value"] =
+          absl::StrJoin(chunks, "().") + "()";
+      method_vars["method_request_body"] = info.body;
+      method_vars["method_http_verb"] = info.http_verb;
+      method_vars["method_rest_path"] =
+          absl::StrCat("absl::StrCat(\"", info.path_prefix, "\",request.",
+                       method_vars["method_request_param_value"], ",\"",
+                       info.path_suffix, "\")");
     }
-    method_vars["method_request_param_value"] =
-        absl::StrJoin(chunks, "().") + "()";
-    method_vars["method_request_body"] = result->body;
-  }
+
+    void operator()(HttpSimpleInfo const& info) {
+      method_vars["method_http_verb"] = info.http_verb;
+      method_vars["method_request_param_value"] = method.full_name();
```

Comment:
why don't we need to do the same `chunks` shenanigans that we do for the `HttpExtensionInfo` case above?